packages/synapse-react-client/src/utils/functions/FileHandleUtils.ts

```diff
@@ -82,38 +79,32 @@ export function getFileHandleStorageInfo(
  */
 export function getStorageLocationName(
   fileHandle: FileHandle,
-  storageLocationUploadDestination?: UploadDestination,
+  storageLocationUploadDestination: UploadDestination | undefined,
 ) {
-  const baseKey =
-    storageLocationUploadDestination &&
-    (storageLocationUploadDestination as BaseKeyUploadDestination).baseKey
-  const baseKeyText = baseKey ? `/${baseKey}` : ''
+  // Uploads to Synapse Storage often do not get their storage location field back-filled,
+  // so null also indicates a Synapse-Stored file
+  if (
+    fileHandle.concreteType === S3_FILE_HANDLE_CONCRETE_TYPE_VALUE &&
+    fileHandle.storageLocationId === null
+  ) {
+    return SYNAPSE_STORAGE
+  }
 
   switch (fileHandle.concreteType) {
     case PROXY_FILE_HANDLE_CONCRETE_TYPE_VALUE:
       return (fileHandle as ProxyFileHandle).filePath
     case EXTERNAL_OBJECT_STORE_FILE_HANDLE_CONCRETE_TYPE_VALUE:
-      return `${
-        (fileHandle as ExternalObjectStoreFileHandle).bucket
-      }${baseKeyText}`
-    case EXTERNAL_FILE_HANDLE_CONCRETE_TYPE_VALUE:
-      return (fileHandle as ExternalFileHandle).externalURL
     case S3_FILE_HANDLE_CONCRETE_TYPE_VALUE:
-      // Uploads to Synapse Storage often do not get their storage location field back-filled,
-      // so null also indicates a Synapse-Stored file
-      if (
-        [null, SYNAPSE_STORAGE_LOCATION_ID].includes(
-          (fileHandle as S3FileHandle).storageLocationId,
-        )
-      ) {
-        return SYNAPSE_STORAGE
-      } else {
-        return `s3://${(fileHandle as S3FileHandle).bucketName}${baseKeyText}`
-      }
     case GOOGLE_CLOUD_FILE_HANDLE_CONCRETE_TYPE_VALUE:
-      return `gs://${
-        (fileHandle as GoogleCloudFileHandle).bucketName
-      }${baseKeyText}`
+      if (storageLocationUploadDestination) {
```

Comment:
Reuse `getUploadDestinationString` when we could get an upload destination using the storage location id. Otherwise, default to the bucket value stored in the file handle.

Reply:
Thanks for this consolidation